server/src/modules/auth/route.ts

```diff
@@ -101,10 +104,16 @@ export default class AuthRoute {
 				error.identifier
 			);
 		}
+
+		const parsed = new URL(returnToUrl);
+		// Use one-shot function for type safety
+		((key: Avo.Auth.IdpLinkedSuccessQueryParam = 'linked') =>
+			parsed.searchParams.append(key, idpUserInfo.type))();
```

Comment:
just define a typed variable:
```suggestion
		const parsed = new URL(returnToUrl);
		const linked: Avo.Auth.IdpLinkedSuccessQueryParam = 'linked';
		parsed.searchParams.append(linked, idpUserInfo.type);
```